web/modules/custom/server_general/tests/src/ExistingSite/ServerGeneralEntityReferecesSelect2Test.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Drupal\Tests\server_general\ExistingSite;
+
+use weitzman\DrupalTestTraits\ExistingSiteBase;
+
+/**
+ * Tests entity references.
+ */
+class ServerGeneralEntityReferecesSelect2Test extends ExistingSiteBase {
+
+  /**
+   * Checks all entity references referencing nodes use the select2 widget.
+   */
+  public function testSelect2Widget() {
+    $em = \Drupal::entityTypeManager();
+    $field_configs = $em->getStorage('field_config')->loadMultiple();
+    $form_display_storage = $em->getStorage('entity_form_display');
+
+    foreach ($field_configs as $id => $field_config) {
+      if ($field_config->getType() !== 'entity_reference') {
+        continue;
+      }
+
+      $storage_definition = $field_config->getFieldStorageDefinition();
+      $referencing = $storage_definition->getSetting('target_type');
+      if ($referencing != 'node') {
+        continue;
+      }
```

Comment:
Let's always check that

```suggestion
      if ($referencing != 'node') {
        continue;
      }
```